release-notes/10.0/preview/preview4/aspnetcore.md

```diff
@@ -2,14 +2,364 @@
+
+<!-- In the docs, highlight the lines with the call to "GetOrCreateSchemaAsync" and "AddComponent" -->
+```csharp
+builder.Services.AddOpenApi(options =>
+{
+    options.AddOperationTransformer(async (operation, context, cancellationToken) =>
+    {
+        // Generate schema for error responses
+        var errorSchema = await context.GetOrCreateSchemaAsync(typeof(ProblemDetails), null, cancellationToken);
+        context.Document?.AddComponent("Error", errorSchema);
+
+        operation.Responses ??= new OpenApiResponses();
+        // Add a "4XX" response to the operation with the newly created schema
+        operation.Responses["4XX"] = new OpenApiResponse
+        {
+            Description = "Bad Request",
+            Content = new Dictionary<string, OpenApiMediaType>
+            {
+                ["application/problem+json"] = new OpenApiMediaType
+                {
+                    Schema = new OpenApiSchemaReference("Error", context.Document)
+                }
+            }
+        };
+    });
+});
+```
+
+## Improved OpenAPI XML documentation support
+
+ASP.NET Core now provides enhanced support for OpenAPI XML documentation comments, particularly for complex types. The updated generator also skips processing problematic complex types, preventing build errors and ensuring smoother builds. Note that this may result in missing metadata in some cases.
+
+You can also now configure XML documentation processing to include comments from other assemblies. This is helpful for documenting types defined outside the current assembly, like `ProblemDetails` from the ASP.NET Core shared framework.
+
+The following example demonstrates how to include XML comments for types in the `Microsoft.AspNetCore.Http` assembly:
+
+```xml
+<Target Name="AddOpenApiDependencies" AfterTargets="ResolveReferences">
+    <ItemGroup>
+        <!-- Include XML documentation for ProblemDetails -->
+        <AdditionalFiles
+                Include="@(ReferencePath->'%(RootDir)%(Directory)%(Filename).xml')"
+                Condition="'%(ReferencePath.Filename)' == 'Microsoft.AspNetCore.Http.Abstractions'"
+                KeepMetadata="Identity;HintPath" />
+    </ItemGroup>
+</Target>
+```
+
+In future previews we expect to include XML comments from key assemblies in the shared framework, reducing the need for manual configuration.
+
+## Access the OpenAPI document with `IOpenApiDocumentProvider`
+
+ASP.NET Core now supports injecting `IOpenApiDocumentProvider` via dependency injection. This enables accessing the OpenAPI document in scenarios like background services or custom middleware, beyond the context of an HTTP request.
+
+```csharp
+public class MyService(IOpenAPiDocumentProvider openApiDocumentProvider)
+{
+    public async Task DoSomething()
+    {
+        OpenApiDocument document = await openApiDocumentProvider.GetOpenApiDocumentAsync();
+    }
+}
+```
+
+## OpenAPI.NET updated to Preview.17
+
```

Comment:
```suggestion
The OpenAPI.NET library used in ASP.NET Core OpenAPI document generation has been upgraded to [v2.0.0-preview.17](https://www.nuget.org/packages/Microsoft.OpenApi/2.0.0-preview.17). This version includes a number of bug fixes and improvements and also introduces some breaking changes. The breaking changes should only impact users that use document, operation, or schema transformers. Breaking changes in this iteration that may impact developers include the following:
```

Reply:
Oops, this went on the wrong line. The intended change was adding a `.` between preview and 17.

Reply:
Got it - thx!